File: src/dataset/cpp/rna-transcription_exercism-viniciusjavs/openAI/ChatGPT4_rna_transcription_v2.cpp
#include "rna_transcription.h"
#include <unordered_map>

namespace rna_transcription {

char to_rna(char dna) {
    switch (dna) {
        case 'G': return 'C';
        case 'C': return 'G';
        case 'T': return 'A';
        case 'A': return 'U';
        default: throw std::invalid_argument("Invalid DNA nucleotide");
    }
}

string to_rna(const string& strand) {
    string rna;
    rna.reserve(strand.size());
    for (char nucleotide : strand) {
        rna.push_back(to_rna(nucleotide));
    }
    return rna;
}

} // namespace rna_transcription

File: src/dataset/javascript/rna-transcription_exercism-elenamountz/openAI/ChatGPT4_rna-transcription_v2.js
export const toRna = (dna) => {
  if (dna === '') return '';
  
  const transcriptionMap = {
    G: 'C',
    C: 'G',
    T: 'A',
    A: 'U'
  };

  return [...dna].map(nucleotide => {
    if (!transcriptionMap[nucleotide]) throw new Error("Invalid input DNA.");
    return transcriptionMap[nucleotide];
  }).join('');
};